Assets/MixedRealityToolkit/Interfaces/Registrars/IMixedRealityServiceRegistrar.cs

```diff
@@ -0,0 +1,143 @@
+    /// </summary>
+    public interface IMixedRealityServiceRegistrar
+    {
+        #region IMixedRealityService registration
+
+        /// <summary>
+        /// Registers a service of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to be registered (ex: IMixedRealityBoundarySystem).</typeparam>
+        /// <param name="serviceInstance">An instance of the service to be registered.</param>
+        bool RegisterService<T>(T serviceInstance) where T : IMixedRealityService;
+
+        /// <summary>
+        /// Registers a service of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to be registered (ex: IMixedRealityBoundarySystem).</typeparam>
+        /// <param name="concreteType">The concrete type to instantiate.</param>
+        /// <param name="supportedPlatforms">The platform(s) on which the service is supported.</param>
+        /// <param name="args">Optional arguments used when instantiating the concrete type.</param>
+        /// <returns>True if the service was successfully registered, false otherwise.</returns>
+        bool RegisterService<T>(
+            Type concreteType,
+            SupportedPlatforms supportedPlatforms = (SupportedPlatforms)(-1),
+            params object[] args) where T : IMixedRealityService;
+
+        /// <summary>
+        /// Unregisters a service of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to be unregistered (ex: IMixedRealityBoundarySystem).
+        /// <param name="name">The name of the service to unregister.</param>
+        /// <returns>True if the service was successfully unregistered, false otherwise.</returns>
+        /// <remarks>If the name argument is not especified, the first instance will be unregistered</remarks>
+        bool UnregisterService<T>(string name = null) where T : IMixedRealityService;
+
+        /// <summary>
+        /// Unregisters a service.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to be unregistered (ex: IMixedRealityBoundarySystem).
+        /// <param name="service">The specific service instance to unregister.</param>
+        /// <returns>True if the service was successfully unregistered, false otherwise.</returns>
+        bool UnregisterService<T>(T serviceInstance) where T : IMixedRealityService;
+
+        /// <summary>
+        /// Checks to see if a service of the specified type has been registered.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service (ex: IMixedRealityBoundarySystem).
+        /// <param name="name">The name of the service.</param>
+        /// <returns>True if the service is registered, false otherwise.</returns>
+        bool IsServiceRegistered<T>(string name = null) where T : IMixedRealityService;
```

Comment:
Similar question to the one above, what's the use case for having a null name (i.e. is this "is any service registered?")

Reply:
Woooooops sent too soon again, followup here is trying to understand the use case for asking that question (i.e. who would use "is any service registered?")

Reply:
It;s a good way to check to see if, for example, the boundary system is running.

The call would look like

```c#
if (registrar.IsServiceRegistered<IMixedRealityBoundarySystem>()) { ... }
```

Reply:
Got it - I think what I was trying to wrap my head around was, I didn't see the other register calls having a name option, so I wasn't sure how this name would be populated to the registrar.